src/session.ts

```diff
@@ -3,15 +3,45 @@ import Assignment from './assignment';
 import AssignmentOverride from './assignmentOverride';
 import TestTrackConfig from './testTrackConfig';
 import Visitor from './visitor';
+import MixPanelAnalytics from './mixpanelAnalytics';
 
-let loaded = null;
+let loaded: null | ((value?: Visitor | PromiseLike<Visitor>) => void) = null;
+
+export type Variants = {
+  [key: string]: () => void;
+};
+
+export type VaryOptions = {
+  variants: Variants;
+  context: string;
+  defaultVariant: boolean | string;
+};
+
+export type AbOptions = {
+  callback: (assignment: boolean | string) => void;
+  context: string;
+  trueVariant: boolean | string;
+  visitor: Visitor;
+};
+
+type SessionOptions = {
+  analytics?: MixPanelAnalytics;
+  errorLogger?: (errorMessage: string) => void;
+  onVisitorLoaded?: Function;
```

Comment:
```suggestion
  onVisitorLoaded?: (visitor: Visitor) => void;
```

or is something else going on here?